Comment 기능 추가

maong0927 opened the pull request:
### 새로운 기능
1. Comment 추가 ( 마크다운 지원 )
2. Comment 삭제 
3. Comment 수정  
4. Comment 한 페이지 보기 
( 아직 페이지네이션 구현을 못했습니다 ㅠ 
원래 무한스크롤로 구현하려했으나 edit 이나 delete 를 할 때의 처리를 시간이 촉박하여 아직 생각하지 못했습니다.... 
발견하는 대로 바로 구현하겠습니다 )

### 작업 유형
- [x] 신규 기능 추가
- [ ] 버그 수정
- [ ] 리펙토링
- [ ] 문서 업데이트

### 체크리스트
- [x] Merge 하는 브랜치가 올바른가?
- [x] 코딩컨벤션을 준수하는가?
- [x] PR과 관련없는 변경사항이 없는가?
- [x] 내 코드에 대한 자기 검토가 되었는가?
- [ ] 변경사항이 효과적이거나 동작이 작동한다는 것을 보증하는 테스트를 추가하였는가?
- [ ] 새로운 테스트와 기존의 테스트가 변경사항에 대해 만족하는가?

Eunbin-Kim committed:
Merge branch 'develop' of https://github.com/boostcamp-2020/Project11-A-Web-FE-Performance-Monitoring-Admin into feature/projectSetting

maong0927 committed:
chore   : react-markdown 설치

1. 댓글 마크다운 지원을 위한 react-markdown 설치

maong0927 committed:
feat    : add comment api 추가

1. comment 를 추가하기 위한 api 추가

maong0927 committed:
feat    : delete comment api 추가

1. comment를 삭제하기 위한 api 추가

maong0927 committed:
feat    : edit comment api 추가

1. comment를 수정하기 위한 api 추가

maong0927 committed:
feat    : get comments api 추가

1. comment들을 가져오는 api 추가

maong0927 committed:
feat    : comment를 입력할 수 있는 textArea 컴포넌트 구현

maong0927 committed:
feat    : comment type 추가

1. comment 객체 type 추가

maong0927 committed:
feat    : markdown이 변환된 모습을 보여주는 컴포넌트 추가

1. 개별 comment 를 보여줄 때 사용
2. comment add할 때 preview 로 사용
3. comment edit할 때 preview 로 사용

maong0927 committed:
feat    : 개별 comment를 보여줄 수 있는 commentBox 컴포넌트 구현

1. 개별 comment 를 보여주는 markdown box 를 가지고 있음.
2. 닉네임, update 날짜, edit 버튼, delete 버튼이 있음.

maong0927 committed:
refactor: 필요없는 스타일 제거

maong0927 committed:
feat    : edit, add 상태 관리 컨테이너 구현

1. markdown 프리뷰와 입력창을 전환할 때 관리되어야 하는 상태들이 있다.
2. 그런 상태들을 관리하기 위해 이 컨테이너를 만들었다.

maong0927 committed:
feat    : commentBox 의 상태를 관리하는 컨테이너 추가

1. 이 커멘트가 현재 수정중인지 아닌지 상태를 관리하는 컨테이너 추가

maong0927 committed:
feat    : 전체 comment 목록 상태를 관리하는 컨테이너 추가

1. 커멘트 렌더링 상태 관리
2. 커멘트 전체 목록 상태 관리

maong0927 committed:
feat    : Comment Tab 추가

1. detail tab 에서 comment tab 추가

maong0927 committed:
fix     : route path 변경

maong0927 committed:
refactor: any 타입 변경

1. event, changeRenderFlip 함수 타입 정의

Eunbin-Kim committed:
feat: ie9 지원을 위한 react-app-polyfill 추가
- ie9에서도 돌아가도록 react-app-polyfill 추가하여 Promise, Object.assign 등 polyfill

Eunbin-Kim committed:
feat: 로그인 시 background image 적용

Eunbin-Kim committed:
feat: sidebar style 조정

Eunbin-Kim committed:
fix: Suspense fallback property 수정
- fallback property Functional Component 선언에 따라 Warning 발생
- lazy loading에 따른 Suspense 컴포넌트에 fallback property 컴포넌트로 선언하는 것으로 수정

Eunbin-Kim committed:
feat: 전체 폰트 조정

maong0927 committed:
feat    : 날짜 형식 변경

1. 날짜 형식 : timestamp -> yy mm dd hh:mm:ss 형식으로 변경

maong0927 committed:
feat    : delete comment 이벤트 추가

1. delete comment 를 클릭하면 comment 가 delete 되는 이벤트 추가

maong0927 committed:
feat    : edit 누를 시 원래 comment 가 text 에 들어가도록 설정

Eunbin-Kim committed:
refactor: Issue page container presenter로 디렉토리 변경
- Issue page 디렉토리 구조 container presenter 구조로 확실히 분리하기 위해 디렉토리 변경
- presenter component는 components 폴더로 넣어 container component와 분리

Eunbin-Kim committed:
refactor: Issue list page container presenter 패턴으로 폴더구조 변경
- Issue list page 또한 presenter 역할을 하므로 component 안으로 이동

maong0927 committed:
fix: textArea 의 값이 markdown preview로 안넘어가는 문제 해결

Eunbin-Kim committed:
feat: New Project Page container presenter 디렉토리 분리 및 컴포넌트 재사용 하도록 리팩토링
- 기존 presentational component 에서 API 요청하던 것 container 컴포넌트로 분리
- ProjectAdmin, ProjectMember 중복 코드 삭제하고 한 presentational component 로 재사용

Eunbin-Kim committed:
refactor: New Project Example page container presenter 패턴으로 디렉토리 구조 변경
- New Project Example은 presenter component이므로 component 디렉토리 안으로 이동

Eunbin-Kim committed:
refactor: Project List page container presenter 패턴 적용
- Project List page presentational component 이므로 component 폴더 안으로 이동
- Simple Card에 있던 Dispatch 상태에 의존하므로 Container 안의 로직으로 이동하여 분리시킴

Eunbin-Kim committed:
refactor: Project Setting Page Conatiner Presenter 분리
- 기존 Presenter component 에 API 호출 섞여있던 것에서, presenter는 view만 담당하고 container에서 API 처리하도록 변경

maong0927 committed:
fix     : 댓글 CRUD 실패 시 alert 설정

gitdog01 commented on src/pages/Issue/components/Comments/CommentHandleContainer.tsx:
```diff
@@ -0,0 +1,161 @@
+    },
+
+    tab: {
+      minWidth: 80, // a number of your choice
+      width: 80, // a number of your choice
```
주석 무슨 의미인가요 ??

maong0927 replied:
지워야할거같습니닿 ㅎ

minbert commented on src/pages/Issue/components/Comments/CommentBox.tsx:
```diff
@@ -0,0 +1,119 @@
+  color: inherit;
+  padding-top: 0;
+  padding-bottom: 0;
+`;
```
EditButton 과 DeleteButton 에서 겹치는 부분은 BaseButton과 같은 변수로 만들어서 styled(BaseButton)``과 같이 쓰는게 좋을 것 같아요.

gitdog01 commented on src/pages/Issue/components/Comments/CommentsTabContainer.tsx:
```diff
@@ -0,0 +1,76 @@
+  const classes = useStyles();
+
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [renderFlip, setRenderFlip] = useState<boolean>(false);
+
+  const changeRenderFlip = () => {
+    setRenderFlip(!renderFlip);
+  };
```
renderFlip 부분이 정확히 어떻게 사용되고 있는 것 인가요 ??

maong0927 replied:
edit, remove, post 후에 다시 렌더링 하기 위해 사용하고 있습니당

Eunbin-Kim replied:
comments, setComments 를 활용해서 상태값 변경에 따라 렌더링 되게 할 수 있을 것 같아요! DB에서 제대로 저장이 안됐을 때의 오류는 API 호출시에 try catch로 잡을 수 있지 않을까 하는 생각이 듭니다...! 같이 더 고민해봐요ㅠㅠ

Eunbin-Kim committed:
fix: route 경로 설정 변경에 따른 경로 수정
- newprojectexample에서 newproject/example 로 수정하여 history push 부분 수정

gitdog01 reviewed (approved):
직접 구성된 모습도 보고싶네요 ~

minbert reviewed (approved):
수고하셨습니다.

Eunbin-Kim commented on src/pages/Issue/components/Comments/CommentBox.tsx:
```diff
@@ -0,0 +1,119 @@
+  changeRenderFlip,
+}: Props) => {
+  const handleDeleteClick = async () => {
+    const result = await deleteComment(commentId);
```
API 호출하는 서비스 코드는 container 단으로 올리면 어떨까요?!

Eunbin-Kim reviewed (approved):
고생 많으셨습니당!!! markdown 적용하다니 대단해여~~

Eunbin-Kim committed:
Merge remote-tracking branch 'origin' into feature/refactor

# Conflicts:
#	package-lock.json
#	src/route.tsx

Eunbin-Kim committed:
Merge branch 'develop' of https://github.com/boostcamp-2020/Project11-A-Web-FE-Performance-Monitoring-Admin into feature/refactor

# Conflicts:
#	package-lock.json
#	src/pages/SignIn/SignIn.tsx

Eunbin-Kim committed:
refactor: alert confirm으로 수정
- 프로젝트 삭제 취소 가능하도록 confirm창으로 수정

Eunbin-Kim committed:
refactor: directory 구조 변경에 따른 import문 수정

Eunbin-Kim committed:
fix: public img 에서 svg 파일 첨부 오류 수정
- public img webpack으로 인해 제대로 파일을 불러오지 못하는 오류 발생
- svg react component 로 선언하여 component 처럼 불러오도록 수정

Eunbin-Kim committed:
feat: image module로 불러오기 위해 d.ts 파일 추가
- webpack에 이미지 파일들 모듈로 불러오도록 선언하였으므로 ts에서도 모듈로 인식하도록 d.ts 파일 추가

Eunbin-Kim committed:
fix: public/img src 지정에서 이미지 파일 module로 import 하도록 수정
- src 지정 시 webpack build 과정에서 src 무너지기 때문에 module로 import 하도록 수정

Eunbin-Kim committed:
refactor: 사용하지 않는 svg 파일 삭제

Eunbin-Kim committed:
refactor: svg 파일 import 경로 수정
- common 폴더에 svg 폴더 생성하여 svg파일들 같이 관리할 수 있도록 수정함

Eunbin-Kim committed:
chore: webpack에 public 디렉토리 alias 추가

Eunbin-Kim committed:
fix: newproject 에 중첩 route 구현함으로써 exact path로 수정
- 기존 /newprojectexample 에서 newproject/example 로 route로 변경하면서 제대로 페이지가 넘어가지 않는 오류 발생
- 두 경로 모두 path로 선언되어 있어 newproject, newproject/example 을 모두 매칭하여 발생한 문제로 exact 로 변경하여 해결

Eunbin-Kim committed:
Merge branch 'feature/comment' of https://github.com/boostcamp-2020/Project11-A-Web-FE-Performance-Monitoring-Admin into feature/comment

# Conflicts:
#	package-lock.json
#	src/pages/Issue/Tags/components/IssueTags.tsx
#	src/pages/Issue/components/IssueTabs.tsx
#	src/route.tsx

Eunbin-Kim committed:
refactor: presenter, container 디렉토리 분리
- presenter, container 컴포넌트 분리하여 경로 수정
